test/Microsoft.AspNet.TestHost.Tests/TestClientTests.cs

```diff
@@ -111,5 +113,38 @@ public TestClientTests()
             // Assert
             Assert.Equal("Hello world POST Response", await response.Content.ReadAsStringAsync());
         }
+
+        [Fact]
+        public async Task ClientDisposalAbortsRequest()
+        {
+            // Arrange
+            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>();
+            RequestDelegate appDelegate = async ctx =>
+            {
+                // Write Headers
+                await ctx.Response.Body.FlushAsync();
+
+                var sem = new SemaphoreSlim(0);
+                try
+                {
+                    await sem.WaitAsync(ctx.RequestAborted);
+                }
+                catch(Exception e)
+                {
+                    tcs.SetException(e);
+                }
+            };
+
+            // Act
+            var server = TestServer.Create(app => app.Run(appDelegate));
+            var client = server.CreateClient();
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:12345");
+            var response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            // Abort Request
+            response.Dispose();
```

Comment:
There should be a separate test for aborting a request before the response is received. You could do this with a CancellationToken or just dispose the HttpClient.